### PR TITLE
Support Vim and Windows

### PR DESCRIPTION
**Pull request by tamago324**
It looked like a great plugin, so I wanted to use it with Vim.
Implemented using the popup feature of Vim.

We also added Windows support.

Thank you!

Refer #8

### CODE REVIEW

File: autoload/blamer.vim [hunk 1]
```diff
@@ -13,10 +13,18 @@ let s:blamer_date_format = get(g:, 'blamer_date_format', '%d/%m/%y %H:%M')
...
+  let s:blamer_namespace = nvim_create_namespace('blamer')
+endif
+if exists('*prop_type_add')
+  let s:prop_type_name = 'blamer_popup_marker'
+endif
 let s:blamer_delay = get(g:, 'blamer_delay', 1000)
 let s:blamer_timer_id = -1
 
+let s:is_windows = has('win16') || has('win32') || has('win64') || has('win95')
+let s:is_not_work = !has('nvim') && !exists('*popup_create')
```
Comment (simonweil) [inline autoload/blamer.vim]: Maybe s:missing_popup_feature ?

```suggestion
let s:missing_popup_feature = !has('nvim') && !exists('*popup_create')
```

Comment (tamago324) [reply autoload/blamer.vim]: Yes!
Thank you

Comment (tamago324) [reply autoload/blamer.vim]: Fixed.

File: autoload/blamer.vim [hunk 2]
```diff
@@ -109,23 +117,68 @@ function! blamer#SetVirtualText(buffer_number, line_number, message) abort
...
+  let l:popup_winid = popup_create(s:blamer_prefix . a:message, {
+  \ 'textprop': 'blamer_popup_marker',
+  \ 'textpropid': l:propid,
+  \ 'line': -1,
+  \ 'col': l:col == 1 ? 1 : 2,
+  \ 'fixed': 1,
+  \ 'wrap': 0,
+  \ 'highlight': 'Blamer'
+  \})
+endfunctio
```
Comment (APZelos) [inline autoload/blamer.vim]: There is a typo here 😄 

```suggestion
endfunction
```

File: autoload/blamer.vim [hunk 3]
```diff
@@ -109,23 +117,68 @@ function! blamer#SetVirtualText(buffer_number, line_number, message) abort
...
+
 function! blamer#Show() abort
-  let l:file_path = expand('%:p')
+  if g:blamer_enabled == 0 || s:is_not_work
```
Comment (APZelos) [inline autoload/blamer.vim]:
```suggestion
  if g:blamer_enabled == 0 || s:missing_popup_feature
```

File: autoload/blamer.vim [hunk 4]
```diff
@@ -163,7 +216,18 @@ function! blamer#Init() abort
     return
   endif
 
-  let l:result = split(system('git rev-parse --show-toplevel 2>/dev/null'), '\n')
+  if s:is_not_work
```
Comment (APZelos) [inline autoload/blamer.vim]:
```suggestion
  if s:missing_popup_feature
```

**Comment by tamago324** [comment]
Thank you review 😂
Fixed.

**Comment by tamago324** [comment]
Thank you for merging!

**Comment by junkblocker** [comment]
Does this really work? In gvim (8.1.2234) I get

```
Error detected while processing function <lambda>9[1]..blamer#Show[23]..blamer#CreatePopup:
line   16:
E475: Invalid argument: blamer_popup_marker
```

**Comment by tamago324** [comment]
@junkblocker How to reproduce it?

**Comment by junkblocker** [comment]
This is on macOS latest.

1. Remove all `.vimrc` , `.vim`  or move them aside etc.
2. `cd && git clone https://github.com/APZelos/blamer.nvim.git .vim`
3. cd .vim
4. gvim
5. `:e README.md`
6. `:BlamerToggle`

**Comment by mozzi0131** [comment]
@junkblocker Hi, did you solve the problem? I met the same error too. 
```
Error detected while processing function <lambda>9[1]..blamer#Show[23]..blamer#CreatePopup:
line   16:
E475: Invalid argument: blamer_popup_marker
```

**Comment by junkblocker** [comment]
@mozzi0131 - I have been running VimR (nvim) instead of MacVim for a while now, so, not sure. Just tried with MacVim and `:BlamerShow` just shows the first line I am on when I switch it on but doesn't move to other lines as I move the cursor...so busted in different ways.

**Comment by tamago324** [comment]
I also tried to get it to work on Neovim on Ubuntu recently, but it didn't work, so when I have time, I'll take a look.

**Comment by mozzi0131** [comment]
@junkblocker @tamago324 Thanks for sharing! It seems like it is not working on vim, Ubuntu 20.04.

**Pull request merged**